### PR TITLE
OFS-14: Add test helpers, improve models

### PR DESCRIPTION
**Pull request by dborowiecki**
Ticket: [OFS-14](https://openimis.atlassian.net/browse/OFS-14) (part of shell backend)
Content of this PR relies on changes from:
* [Contribution Plan PR](https://github.com/openimis/openimis-be-contribution_plan_py/pull/2)
* [Policy Holder PR](https://github.com/openimis/openimis-be-policyholder_py/pull/1)

### CODE REVIEW

File: contract/gql_mutations/create_contract.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        if type(user) is AnonymousUser or not user.id:
+            raise ValidationError("mutation.authentication_required")
+        # if not user.has_perms(ContractConfig.gql_mutation_create_contract_perms):
+        #     raise PermissionDenied(_("unauthorized"))
+
+    @classmethod
+    def _mutate(cls, user, **data):
+        data['policy_holder'] = user.id_for_audit
+        data['status'] = Contract.STATUS_ENTERED  # TODO: Determine all possible statuses for contract ConfigEnum
+        data['validity_from'] = TimeUtils.now()
```
Comment (delcroip) [inline contract/gql_mutations/create_contract.py:28]: I don't want to keep this beaviour(UUIDVersionnedModel), validityTo/ValidityFrom was repalced by ContractTo/ContractFrom that are for business logic (Contract for dec is from 1/12 to 31/12)

Comment (delcroip) [reply contract/gql_mutations/create_contract.py:28]: But it is valid for DateUpdated/Created

File: contract/gql_mutations/delete_contract.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    _mutation_class = "DeleteContractMutation"
+
+    class Input(ContractInputType):
+        pass
+
+    @classmethod
+    def async_mutate(cls, user, **data):
+        output = []
+        for contract_uuid in data["uuids"]:
+            deletion_result = super(DeleteContractMutation, cls).async_mutate(user, contract_uuid=contract_uuid)
```
Comment (delcroip) [inline contract/gql_mutations/delete_contract.py:20]: I might have a flaw in my logic (using history) for deleted object, we might need to add a member isActive/IsDeleted to UUIDHistoryModel so this flag is changed when a row is "Deleted"

**Review by delcroip — CHANGES_REQUESTED** [review]

**Comment by delcroip** [comment]
need to have the discussion with BS on the base_mutation as discussed this morning

File: contract/gql_mutations/update_contract.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+    class Input(ContractInputType):
+        pass
+
+    @classmethod
+    def _validate_mutation(cls, user, **data):
+        if type(user) is AnonymousUser or not user.id:
+            raise ValidationError("mutation.authentication_required")
+
+        contract_uuid = data['uuid']
```
Comment (dragos-dobre) [inline contract/gql_mutations/update_contract.py:20]: You suppose to have uuid in the data.

**Review by dragos-dobre — CHANGES_REQUESTED** [review]

File: contract/gql_mutations/update_contract.py (hunk 3 above)
Comment (delcroip) [reply contract/gql_mutations/update_contract.py:20]: mutation data has all the data sent by fe if I am not mistaken

**Comment by dborowiecki** [comment]
Closing PR as outdated, included functionalities present in [PR#3](https://github.com/openimis/openimis-be-contract_py/pull/3)

**Pull request closed without merging**